cranelift-codegen/src/isa/x86/abi.rs

```diff
@@ -331,71 +224,72 @@ pub fn legalize_signature(
         shared_flags,
         isa_flags,
     );
+    let mut rets2 = rets.clone();
```

Comment:
It's unfortunate the clone happens even when one doesn't use multi-value returns. To make this slightly more zero-cost, could we clone only when `sig.is_multi_return` instead?